packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -217,59 +220,84 @@ class _TimePickerHeader extends StatelessWidget {
+      case ThemeBrightness.light:
+        backgroundColor = themeData.primaryColor;
+        break;
+      case ThemeBrightness.dark:
+        backgroundColor = themeData.backgroundColor;
+        break;
+    }
+
+    TextStyle activeStyle = headerTextTheme.display3.copyWith(
+      fontSize: _kHeaderFontSize, color: activeColor
+    );
+    TextStyle inactiveStyle = headerTextTheme.display3.copyWith(
+      fontSize: _kHeaderFontSize, color: inactiveColor
+    );
 
     TextStyle hourStyle = mode == _TimePickerMode.hour ? activeStyle : inactiveStyle;
     TextStyle minuteStyle = mode == _TimePickerMode.minute ? activeStyle : inactiveStyle;
 
-    TextStyle amStyle = headerTheme.subhead.copyWith(
+    TextStyle amStyle = headerTextTheme.subhead.copyWith(
       color: selectedTime.period == DayPeriod.am ? activeColor: inactiveColor
     );
-    TextStyle pmStyle = headerTheme.subhead.copyWith(
+    TextStyle pmStyle = headerTextTheme.subhead.copyWith(
       color: selectedTime.period == DayPeriod.pm ? activeColor: inactiveColor
     );
 
     return new Container(
-      padding: const EdgeInsets.fromLTRB(24.0, 24.0, 24.0, 20.0),
-      decoration: new BoxDecoration(backgroundColor: theme.primaryColor),
+      height: 100.0,
+      padding: const EdgeInsets.symmetric(horizontal: 24.0),
+      decoration: new BoxDecoration(backgroundColor: backgroundColor),
       child: new Row(
         children: <Widget>[
-          new GestureDetector(
-            onTap: () => _handleChangeMode(_TimePickerMode.hour),
-            child: new Text(selectedTime.hourOfPeriodLabel, style: hourStyle)
+          new Flexible(
+            child: new Align(
+              alignment: FractionalOffset.centerRight,
+              child: new GestureDetector(
+                onTap: () => _handleChangeMode(_TimePickerMode.hour),
+                child: new Text(selectedTime.hourOfPeriodLabel, style: hourStyle)
+              )
+            )
           ),
           new Text(':', style: inactiveStyle),
-          new GestureDetector(
-            onTap: () => _handleChangeMode(_TimePickerMode.minute),
-            child: new Text(selectedTime.minuteLabel, style: minuteStyle)
-          ),
-          new GestureDetector(
-            onTap: _handleChangeDayPeriod,
-            behavior: HitTestBehavior.opaque,
-            child: new Container(
-              padding: const EdgeInsets.only(left: 16.0, right: 24.0),
-              child: new Column(
+          new Flexible(
+            child: new Align(
+              alignment: FractionalOffset.centerLeft,
+              child: new Row(
                 children: <Widget>[
-                  new Text('AM', style: amStyle),
-                  new Container(
-                    padding: const EdgeInsets.only(top: 4.0),
-                    child: new Text('PM', style: pmStyle)
+                  new GestureDetector(
+                    onTap: () => _handleChangeMode(_TimePickerMode.minute),
```

Comment:
I'm clearly not contributing anything concrete this morning, but. I wish we would reserve the `=>` notation for one-liners that return a value.